src/TestUtils/src/UITest.Appium/Actions/AppiumAndroidAlertActions.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Collections.ObjectModel;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Appium;
+using UITest.Core;
+
+namespace UITest.Appium
+{
+	public class AppiumAndroidAlertActions : ICommandExecutionGroup
+	{
+		const string GetAlertsCommand = "getAlerts";
+		const string GetAlertButtonsCommand = "getAlertButtons";
+		const string GetAlertTextCommand = "getAlertText";
+
+		readonly List<string> _commands = new()
+		{
+			GetAlertsCommand,
+			GetAlertButtonsCommand,
+			GetAlertTextCommand,
+		};
+		readonly AppiumApp _appiumApp;
+
+		public AppiumAndroidAlertActions(AppiumApp appiumApp)
+		{
+			_appiumApp = appiumApp;
+		}
+
+		public bool IsCommandSupported(string commandName)
+		{
+			return _commands.Contains(commandName, StringComparer.OrdinalIgnoreCase);
+		}
+
+		public CommandResponse Execute(string commandName, IDictionary<string, object> parameters)
+		{
+			return commandName switch
+			{
+				GetAlertsCommand => GetAlerts(parameters),
+				GetAlertButtonsCommand => GetAlertButtons(parameters),
+				GetAlertTextCommand => GetAlertText(parameters),
+				_ => CommandResponse.FailedEmptyResponse,
+			};
+		}
+
+		CommandResponse GetAlerts(IDictionary<string, object> parameters)
+		{
+			var alerts = _appiumApp.Query.ById("parentPanel");
+
+			if (alerts is null || alerts.Count == 0)
+				return CommandResponse.FailedEmptyResponse;
+
+			return new CommandResponse(alerts, CommandResponseResult.Success);
+		}
+
+		CommandResponse GetAlertButtons(IDictionary<string, object> parameters)
+		{
+			var alert = GetAppiumElement(parameters["element"]);
+			if (alert is null)
+				return CommandResponse.FailedEmptyResponse;
+
+			var items = AppiumQuery.ByClass("android.widget.ListView")
+				.FindElements(alert, _appiumApp)
+				.FirstOrDefault()
+				?.ByClass("android.widget.TextView");
+
+			var buttons = AppiumQuery.ByClass("android.widget.Button")
+				.FindElements(alert, _appiumApp);
```

Comment:
Not 100% sure this is the _best_ way, but I consider all the "buttons" to be the items to pick and the "extra" buttons like cancel/destroy. This is easier because on iOS they aer all buttons. We _can_ write code to break it out, but why do extra work if in most cases you want to see what all the options are.